### PR TITLE
feat(core): expose cached prompt tokens in ChatUsage

### LINKED ISSUE #1854: [Feature]: Expose cached prompt tokens in ChatUsage (OpenAI prompt_tokens_details.cached_tokens)
## Summary

`ChatUsage` (`io.agentscope.core.model.ChatUsage`) currently exposes only `inputTokens`, `outputTokens`, and `time`. The OpenAI API returns a finer breakdown under `usage.prompt_tokens_details.cached_tokens`, which reports how many of the prompt tokens were served from the prompt cache. This value is **already deserialized** by the framework into `OpenAIUsage.PromptTokensDetails.cachedTokens`, but it is **dropped** when `OpenAIResponseParser` converts `OpenAIUsage` into `ChatUsage`.

As a result, downstream consumers reading `ModelCallEndEvent.getUsage()` (or `Msg.getUsage()`) have no way to know how many input tokens hit the cache, which is essential for accurate cost accounting (cached input tokens are billed at a reduced rate).

## Current Behavior

- `OpenAIUsage.PromptTokensDetails.cachedTokens` is correctly parsed from `prompt_tokens_details.cached_tokens`.
- `OpenAIResponseParser#parseCompletionResponse` and `#parseChunkResponse` build `ChatUsage` using only `getPromptTokens()` and `getCompletionTokens()`; the cached-token detail is discarded.
- `ChatUsage` has no field to carry cached tokens, so the information is unavailable to any consumer of `ModelCallEndEvent` / `ChatResponse`.

## Expected Behavior

`ChatUsage` should carry the cached prompt token count so consumers can compute cache-aware costs.

Note on semantics: `cachedTokens` is a **subset of** `inputTokens` (i.e. `inputTokens = non-cached prompt tokens + cachedTokens`), not an additional amount. `getTotalTokens()` semantics (`inputTokens + outputTokens`) should remain unchanged.

## Proposed Change

1. Add an `int cachedTokens` field to `ChatUsage` with a `getCachedTokens()` getter and a `cachedTokens(int)` builder method.
   - Keep the existing 3-arg constructor (delegating with `cachedTokens = 0`) to preserve backward compatibility for the ~8 existing `ChatUsage.builder()` call sites.
   - Move `@JsonCreator` to the new 4-arg constructor; since the field is a primitive `int`, older serialized JSON without the property deserializes safely to `0` (verified against the expectations in `MsgUsageSerializationTest`).
2. In `OpenAIResponseParser`, add a `getSafeCachedTokens(OpenAIUsage)` helper and populate `.cachedTokens(...)` in both `parseCompletionResponse` and `parseChunkResponse`.
3. Add unit tests for: cached tokens present, absent (`prompt_tokens_details` null → 0), and `ChatUsage` JSON round-trip with the new field.

## Scope

This issue intentionally covers the **OpenAI path only**, since the data is already available at the DTO layer there. Extending cache-token reporting to other providers (e.g. Anthropic `cache_read_input_tokens` / `cache_creation_input_tokens`, which are not yet parsed) can be handled in separate follow-up issues/PRs to keep changes focused.

## Backward Compatibility

- Existing `ChatUsage.builder()` callers are unaffected (`cachedTokens` defaults to `0`).
- Existing serialized `ChatUsage` JSON remains deserializable.
- `getTotalTokens()` behavior is unchanged.

I'd be happy to submit a PR for this if it aligns with the project roadmap.

### PR DESCRIPTION
**Pull request by DennisWLX**
Add cachedTokens field to ChatUsage to carry OpenAI's prompt_tokens_details.cached_tokens through to consumers.

- Add int cachedTokens field with getter and builder method
- Keep backward-compatible 3-arg constructor (defaults to 0)
- Add getSafeCachedTokens() helper in OpenAIResponseParser
- Populate cachedTokens in both non-stream and stream paths
- Add unit tests for all null/absent/present scenarios

Closes #1854

## AgentScope-Java Version

[The version of AgentScope-Java you are working on, e.g. 1.0.12, check your pom.xml dependency version or run `mvn dependency:tree | grep agentscope-parent:pom`(only mac/linux)]

## Description

[Please describe the background, purpose, changes made, and how to test this PR]

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has been formatted with `mvn spotless:apply`
- [ ]  All tests are passing (`mvn test`)
- [ ]  Javadoc comments are complete and follow project conventions
- [ ]  Related documentation has been updated (e.g. links, examples, etc.)
- [ ]  Code is ready for review

### CODE REVIEW

**Review by AgentScopeJavaBot — COMMENTED** [review]
🤖 AI Review

File: agentscope-core/src/main/java/io/agentscope/core/model/ChatUsage.java [hunk 1]
```diff
@@ -22,28 +22,48 @@
...
+        this(inputTokens, outputTokens, 0, time);
+    }
+
     /**
      * Creates a new ChatUsage instance.
      *
      * @param inputTokens the number of tokens used for the input/prompt
      * @param outputTokens the number of tokens used for the output/generated response
+     * @param cachedTokens the number of input tokens served from the prompt cache (a subset of
+     *     {@code inputTokens}); {@code 0} when the provider does not report cache information
```
Comment (AgentScopeJavaBot) [inline agentscope-core/src/main/java/io/agentscope/core/model/ChatUsage.java:55]: [major] `ReasoningContext` maintains `inputTokens`/`outputTokens`/`time` accumulator fields and builds a final `ChatUsage` via `getChatUsage()`, but the new `cachedTokens` field is not propagated through `processChunk()` or the builder call. In streaming scenarios, `cachedTokens` will always be 0. Add a `cachedTokens` field to `ReasoningContext` and wire it through `processChunk()` and `getChatUsage()` following the same pattern as `inputTokens`.

Comment (AgentScopeJavaBot) [inline agentscope-core/src/main/java/io/agentscope/core/model/ChatUsage.java:55]: [major] `ReActAgent.collectAggregatedUsage()` accumulates `inputTokens`/`outputTokens`/`time` across multi-round structured output but does not accumulate `cachedTokens`. Add `totalCached` to the loop and include it in the final builder call.

Comment (DennisWLX) [reply agentscope-core/src/main/java/io/agentscope/core/model/ChatUsage.java:55]: Good catch, thanks. Fixed in d27cda57.

Added a totalCached accumulator to the loop in collectAggregatedUsage() and included .cachedTokens(totalCached) in the final builder, so cached tokens are now summed across multi-round structured-output calls.

Comment (DennisWLX) [reply agentscope-core/src/main/java/io/agentscope/core/model/ChatUsage.java:55]: Good catch, thanks. Fixed in d27cda57.

Added a cachedTokens field to ReasoningContext and wired it through processChunk() (reads usage.getCachedTokens()) and getChatUsage(). I also found that buildFinalMessage() rebuilds ChatUsage independently from the accumulator fields, so it had the same gap — that path is fixed too. Added ReasoningContextTest.testCachedTokensPropagation to verify cached tokens survive the full streaming path.

**Review by chickenlj — APPROVED** [review]
